les/backend.go

```diff
@@ -80,6 +80,14 @@ type LightEthereum struct {
 	wg sync.WaitGroup
 }
 
+// lesCommons contains needed by both server and client.
+type lesCommons struct {
+	config                       *eth.Config
+	chainDb                      ethdb.Database // Block chain database
+	protocolManager              *ProtocolManager
+	chtIndexer, bloomTrieIndexer *core.ChainIndexer
+}
```

Comment:
Could you separate out the `lesCommons` struct and its methods into a `commons.go` file (perhaps also the `NodeInfo`)? The functions seem to be scattered all over the file here and it's hard to follow.

Reply:
TBH they're just all at the end of the file ;)